leakcanary-android/src/main/java/com/squareup/leakcanary/internal/LeakCanarySingleThreadFactory.java

```diff
@@ -0,0 +1,30 @@
+package com.squareup.leakcanary.internal;
+
+import java.util.concurrent.ThreadFactory;
+
+/**
+ * Copied from {@link java.util.concurrent.Executors#defaultThreadFactory()}. This is intended to
+ * only be used with a single thread executor.
+ */
+final class LeakCanarySingleThreadFactory implements ThreadFactory {
```

Comment:
I'm OK with this. I only pulled in this implementation to keep it as close to what is currently being used as possible.